Jenkinsfile

```diff
@@ -4,7 +4,14 @@ node {
   try {
     // Checkout the proper revision into the workspace.
     stage('checkout') {
-      checkout scm
+      checkout([
+        $class: 'GitSCM',
+        branches: scm.branches,
+        doGenerateSubmoduleConfigurations: scm.doGenerateSubmoduleConfigurations,
```

Comment:
Have you tried this without `doGenerateSubmoduleConfigurations` and `submoduleCfg`? It seems like those parameters are not relevant to the behavior change we need here.